core/device_impl.cpp

```diff
@@ -443,6 +444,40 @@ void DeviceImpl::get_param_float_async(const std::string &name,
                                             callback));
 }
 
+MavlinkCommands::Result DeviceImpl::set_flight_mode(uint8_t custom_sub_mode, uint8_t custom_mode)
+{
+    uint8_t flag_safety_armed = DeviceImpl::is_armed() ? MAV_MODE_FLAG_SAFETY_ARMED : 0;
```

Comment:
`is_armed()` doesn't require explicit qualification, as it's a non-static method of `DeviceImpl`.